lib/widget/calendar.dart

```diff
@@ -0,0 +1,474 @@
+import 'dart:collection';
+
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/framework/model.dart';
+import 'package:ensemble/framework/scope.dart';
+import 'package:ensemble/framework/view/data_scope_widget.dart';
+import 'package:ensemble/framework/widget/widget.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble/widget/helpers/controllers.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/material.dart';
+import 'package:table_calendar/table_calendar.dart';
+import 'package:yaml/yaml.dart';
+
+final kToday = DateTime.now();
+final kFirstDay = DateTime(kToday.year, kToday.month - 3, kToday.day);
+final kLastDay = DateTime(kToday.year, kToday.month + 3, kToday.day);
+
+class EnsembleCalendar extends StatefulWidget
+    with Invokable, HasController<CalendarController, CalendarState> {
+  static const type = 'Calendar';
+  EnsembleCalendar({Key? key}) : super(key: key);
+
+  final CalendarController _controller = CalendarController();
+  @override
+  CalendarController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => CalendarState();
+
+  @override
+  Map<String, Function> getters() {
+    return {
+      'selectedCell': () => _controller.selectedDays.value.toList(),
+      'markedCell': () => _controller.markedDays.value.toList(),
+    };
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {
+      'markCall': (value) => _markCell(value),
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'rowHeight': (value) =>
```

Comment:
This maybe a problem as the row height should expand to accomodate items. Can it just size to its content?

Reply:
The package we are using has imposed this constrains on the cell with SizedBox.